src/Avalonia.Input/ContentInputElement.cs

```diff
@@ -0,0 +1,787 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Avalonia.Controls;
+using Avalonia.Controls.Metadata;
+using Avalonia.Input.GestureRecognizers;
+using Avalonia.Input.TextInput;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    [PseudoClasses(":disabled", ":focus", ":focus-visible", ":pointerover")]
```

Comment:
Just noticed, that ":focus-within" is missed here and on InputElement. Seems like an old issue though.

Reply:
Yeah, ContentInputElement is a bit of a messy copy&paste job since it's pretty much the same implementation as InputElement.
Using composition instead to reduce duplication doesn't really reduce complexity here I think.
Other than that, we don't have multiple inheritance or traits to fix it.
Default methods on the interface don't seem sensible either due to how C# handles them.